### DISCUSSION TITLE
Automate td_state.json

### DISCUSSION THREAD

**Discussion by usapkota** [areed1192/td-ameritrade-api#6] [General]
Is there a way to automatically refresh td_state.json? I am running my trade robot as a docker container in Amazon EC2, but I have to recreate a new docker image with a refreshed td_state.json when the token used to create image expires. When I run my code in docker container it does not update td_state.json file in the image (dont think it is possible to do so). May be not running as a docker container is one option, but wondering if there are other solutions.

**Comment by areed1192** [comment]
Are you talking about when your access token expires or your refresh token?

**Reply by usapkota** [reply]
td_state.json is updated, but my issue was with running it as a docker container. Because the container was not updating the file 'td_state.sjon' within the image. I think the question is more related to docker than this package. I have decided to run without docker, which will solve my problem and keep things simple (which not sure why I was not doing).  My question deserves to be deleted, could not figure out that as well. Thank you for your great work!!

**Reply by areed1192** [reply]
Well, you could still use docker because the way the library is written you can pass through a dictionary to the `TdCredentials` object and just have it refresh the state every time the token expires. All you would need to do is pass through that dictionary when you create the docker image. I would say store is as an environment variable and then inside your python script just load the environment variable and pass it through using the `from_dict` method. Here is a short example:

```python
td_credential = TdCredentials(client_id='', redirect_uri='')
td_credential.from_dict(
       token_dict={
        'access_token': '',
        'refresh_token': ',
        'scope': '',
        'expires_in': 0,
        'refresh_token_expires_in': 0,
        'token_type': ''
    }
)
```
